ci/dev-env-install.sh

```diff
@@ -38,13 +38,18 @@ step "Building dev-env dependencies"
 # repeat this a few times. There does not seem to be an option that we can pass to nix
 # to make it retry itself. See https://github.com/NixOS/nix/issues/2794 for the issue requesting
 # this feature.
+NIX_FAILED=0
 for i in `seq 10`; do
```

Comment:
I'm sceptical this complexity is worth it. I'd just retry 10 times, stopping on the first success or failing if we require all 10 times. Something as simple as `$A || $A || $A || $A || $A || $A || $A || $A` would make me happier - shell script should be feared.

Reply:
If we retry 10 times and have e.g. an actual build failure in GHC, I have to wait for 10 hours until my build actually fails. As much as I dislike shell scripting, I still prefer it over that.

Reply:
You're always in a hurry... Fair enough, that makes sense.

Reply:
I prefer spending my time on more exiting things that build failures :wink: